### PR TITLE
Bugfix : Added `mathjax_ignore` to HTMLFormatter._write_table and changed tests to assert `mathjax_ignore`

### LINKED ISSUE #59884: BUG: Disabling pandas option display.html.use_mathjax has no effect
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
pd.set_option('display.html.use_mathjax', False)
pd.DataFrame([
    "$2.13 dont_make_this_math $5.24"
])
```

### Issue Description

My pandas data frames contain text columns which contain dollar signs. I need to be represented as raw strings not as a math-formatted picture.  

Pandas provides the option `display.html.use_mathjax` to this end. However, setting the option does not change pandas behavior. Even with the config set the string "$2.13 dont_make_this_math $5.24" has its spaces removed and text after `_` subscripted.

I understand this can be managed by explicitly escaping dollar signs in any text column I come across, but this is a lot of overhead.

### Expected Behavior

Screenshot 2024-09-25 at 10 52 27

not as

Screenshot 2024-09-25 at 10 52 18

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : 0691c5cf90477d3503834d983f69350f250a6ff7
python                : 3.11.9
python-bits           : 64
OS                    : Darwin
OS-release            : 23.6.0
Version               : Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6000
machine               : arm64
processor             : arm
byteorder             : little
LC_ALL                : None
LANG                  : en_US.UTF-8
LOCALE                : en_US.UTF-8

pandas                : 2.2.3
numpy                 : 1.26.4
pytz                  : 2024.1
dateutil              : 2.9.0.post0
pip                   : 23.3.1
Cython                : None
sphinx                : None
IPython               : 8.27.0
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.3
blosc                 : None
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : None
html5lib              : None
hypothesis            : None
gcsfs                 : None
jinja2                : 3.1.4
lxml.etree            : None
matplotlib            : 3.9.2
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : None
pandas_gbq            : None
psycopg2              : None
pymysql               : None
pyarrow               : 15.0.2
pyreadstat            : None
pytest                : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : None
sqlalchemy            : None
tables                : None
tabulate              : None
xarray                : None
xlrd                  : None
xlsxwriter            : None
zstandard             : None
tzdata                : 2024.1
qtpy                  : None
pyqt5                 : None
</details>

### PR DESCRIPTION
**Pull request by Maru5er**
- [x] closes #59884 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Feedback would be appreciated.

### CODE REVIEW

**Review by attack68 — APPROVED** [review]
LGTM - test cases covered

**Comment by rhshadrach** [comment]
@datapythonista - the `/preview` feature seems broken, I get a 500 response from the server.

https://github.com/pandas-dev/pandas/actions/runs/11469095078/job/31915606006

Should I open an issue?

File: pandas/io/formats/style_render.py [hunk 1]
```diff
@@ -366,9 +366,11 @@ def _translate(
         if not get_option("styler.html.mathjax"):
             table_attr = table_attr or ""
             if 'class="' in table_attr:
-                table_attr = table_attr.replace('class="', 'class="tex2jax_ignore ')
+                table_attr = table_attr.replace(
+                    'class="', 'class="tex2jax_ignore mathjax_ignore"'
```
Comment (rhshadrach) [inline pandas/io/formats/style_render.py:370]: This should be

    'class="', 'class="tex2jax_ignore mathjax_ignore '

as we are prepending to an already existing class.

File: pandas/io/formats/style_render.py [hunk 2]
```diff
@@ -366,9 +366,11 @@ def _translate(
...
+                )
             else:
-                table_attr += ' class="tex2jax_ignore"'
+                table_attr += ' class="tex2jax_ignore mathjax_ignore"'
```
Comment (rhshadrach) [inline pandas/io/formats/style_render.py:373]: Similar here.

File: doc/source/whatsnew/v3.0.0.rst [hunk 3]
```diff
@@ -698,6 +698,7 @@ I/O
 - Bug in :meth:`read_stata` raising ``KeyError`` when input file is stored in big-endian format and contains strL data. (:issue:`58638`)
 - Bug in :meth:`read_stata` where extreme value integers were incorrectly interpreted as missing for format versions 111 and prior (:issue:`58130`)
 - Bug in :meth:`read_stata` where the missing code for double was not recognised for format versions 105 and prior (:issue:`58149`)
+- Bug in :meth:`set_option` where disabling pandas option display.html.use_mathjax has no effect (:issue:`59884`)
```
Comment (rhshadrach) [inline doc/source/whatsnew/v3.0.0.rst:701]:
```suggestion
- Bug in :meth:`set_option` where setting the pandas option ``display.html.use_mathjax`` to ``False`` has no effect (:issue:`59884`)
```

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style_render.py [hunk 4]
```diff
@@ -366,9 +366,11 @@ def _translate(
         if not get_option("styler.html.mathjax"):
             table_attr = table_attr or ""
             if 'class="' in table_attr:
-                table_attr = table_attr.replace('class="', 'class="tex2jax_ignore ')
+                table_attr = table_attr.replace(
+                    'class="', 'class="tex2jax_ignore mathjax_ignore '
+                )
             else:
-                table_attr += ' class="tex2jax_ignore"'
+                table_attr += 'class="tex2jax_ignore mathjax_ignore"'
```
Comment (attack68) [inline pandas/io/formats/style_render.py:373]: There is a reason for this space at the start.

Comment (attack68) [inline pandas/io/formats/style_render.py:373]:
```suggestion
                table_attr += ' class="tex2jax_ignore mathjax_ignore"'
```

**Comment by KevsterAmp** [comment]
This PR seems to have gone stale, can I take over this issue? Thanks

@rhshadrach @attack68

**Comment by rhshadrach** [comment]
@KevsterAmp - sounds good.

File: pandas/io/formats/style_render.py [hunk 5]
```diff
@@ -366,9 +366,11 @@ def _translate(
         if not get_option("styler.html.mathjax"):
             table_attr = table_attr or ""
             if 'class="' in table_attr:
-                table_attr = table_attr.replace('class="', 'class="tex2jax_ignore ')
+                table_attr = table_attr.replace(
+                    'class="', ' class="tex2jax_ignore mathjax_ignore '
```
Comment (rhshadrach) [inline pandas/io/formats/style_render.py:370]:
```suggestion
                    'class="', 'class="tex2jax_ignore mathjax_ignore '
```

**Review by rhshadrach — CHANGES_REQUESTED** [review]

**Comment by KevsterAmp** [comment]
@rhshadrach - Do I commit on this branch or I re-open a new PR? Thanks

**Comment by mroeschke** [comment]
Thanks for this PR, but the issue was closed by https://github.com/pandas-dev/pandas/pull/60311

**Pull request closed without merging**